addons/im_livechat/static/tests/embed/composer_patch_tests.js

```diff
@@ -0,0 +1,29 @@
+/* @odoo-module */
+
+import { startServer } from "@bus/../tests/helpers/mock_python_environment";
+
+import { loadDefaultConfig, start } from "@im_livechat/../tests/embed/helper/test_utils";
+
+import { triggerHotkey } from "@web/../tests/helpers/utils";
+import { click, contains, insertText } from "@web/../tests/utils";
+
+QUnit.module("composer patch");
+
+QUnit.test("File upload only available when authorized", async () => {
+    const pyEnv = await startServer();
+    await loadDefaultConfig();
+    const env = await start();
+    await click(".o-livechat-LivechatButton");
+    await contains(".o-mail-ChatWindow");
+    await insertText(".o-mail-Composer-input", "Hello, please enable file uploading!");
+    triggerHotkey("Enter");
+    await contains(".o-mail-Message-content", { text: "Hello, please enable file uploading!" });
+    await contains(".o-mail-Composer-attachFiles", { count: 0 });
+    pyEnv.withUser(pyEnv.adminUserId, () =>
+        env.services.orm.call("discuss.channel", "toggle_allow_public_upload", [
+            env.services["im_livechat.livechat"].thread.id,
+        ])
+    );
```

Comment:
Could potentially use multi-tab to avoid fake RPC?

Reply:
Not possible, this environment only includes embed live chat.